fuse_graphs/src/hash_graph.cpp

```diff
@@ -409,6 +409,22 @@ ceres::Solver::Summary HashGraph::optimize(const ceres::Solver::Options& options
   return summary;
 }
 
+void HashGraph::print(std::ostream& stream) const
+{
+  stream << "HashGraph\n"
+         << "  constraints:\n";
+  for (const auto& constraint : constraints_)
```

Comment:
For now I simply print the constraints and variables, but I'm open to suggestions regarding what we print here. There's another map that has variables and constraints pairs that could be printed as well, either replacing what it's printed now, or maybe printing all (constraints list + variables list + variables to constraints list mapping).

Reply:
The variables and constraints are what I typically print as well. If anything, I'm wondering if this implementation should get moved to the `Graph` base class. We can use `getConstraints()` and `getVariables()` to access the collections in a generic way.

Reply:
I think that could be the generic implementation in the base class, but I think we should still allow the user to override the default/base implementation, which I think even allows extending it (by calling the base class implementation explicitly from the child class implementation).